File: 977. Squares of a Sorted Array/Approach03.py
from typing import *

class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        n = len(nums)
        i, j, k = 0, n-1, n-1
        result = [0]*n

        while(k>=0):
            a = nums[i]*nums[i]
            b = nums[j]*nums[j]
            if(a>b):
                result[k] = a
                i += 1
            else:
                result[k] = b
                j -= 1
            k -= 1
        return result

File: 977. Squares of a Sorted Array/Approach03.cpp
#include <vector>
#include <algorithm>

using namespace std;

#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int n = nums.size();
        int i = 0; 
        int j = n-1;
        int k = n-1;
        vector<int> result(n);
        while(k>=0){
            int a = nums[i]*nums[i];
            int b = nums[j]*nums[j];
            if(a > b){
                result[k] = a;
                i++;
            }
            else{
                result[k] = b;
                j--;
            }
            k--;
        }
        
        return result;
    }
};
